Watchman.Discord/Areas/Protection/Controllers/MuteUserController.cs

```diff
@@ -0,0 +1,122 @@
+        [DiscordCommand("mute")]
+        public void MuteUser(DiscordRequest request, Contexts contexts)
+        {
+            var requestParser = new MuteRequestParser(request, _usersService, contexts);
+            var userToMute = requestParser.GetUser();
+            var muteEvent = requestParser.GetMuteEvent(userToMute.Id, contexts);
+
+            MuteUserOrOverwrite(contexts, muteEvent, userToMute).Wait();
+            UnmuteInFuture(contexts, muteEvent, userToMute);
+        }
+
+        //[IgnoreForHelp] todo:
+        [DiscordCommand("unmute")]
+        public void UnmuteUser(DiscordRequest request, Contexts contexts)
+        {
+            var requestParser = new MuteRequestParser(request, _usersService, contexts);
+            var userToUnmute = requestParser.GetUser();
+
+            UnmuteUserOnlyIfMuted(contexts, userToUnmute);
+        }
+
+        private async Task MuteUserOrOverwrite(Contexts contexts, MuteEvent muteEvent, UserContext userToMute)
+        {
+            var possiblePreviousUserMuteEvent = GetNotUnmutedUserMuteEvent(contexts.Server, userToMute);
+            if (possiblePreviousUserMuteEvent != null)
+            {
+                var markAsUnmuted = new MarkMuteEventAsUnmutedCommand(possiblePreviousUserMuteEvent);
+                await _commandBus.ExecuteAsync(markAsUnmuted);
+            }
+
+            await _muteService.MuteUser(userToMute, contexts.Server);
+            await _commandBus.ExecuteAsync(new AddMuteEventToBaseCommand(muteEvent));
+
+            var messagesService = _messagesServiceFactory.Create(contexts);
+            await messagesService.SendResponse(x => x.MutedUser(userToMute, muteEvent.TimeRange.End), contexts);
+        }
+
+        private MuteEvent? GetNotUnmutedUserMuteEvent(DiscordServerContext server, UserContext userContext)
+        {
+            var userMuteEvents = GetMuteEvents(server, userContext.Id);
+            // in the same time there should exists only one MUTED MuteEvent per user per server
+            return userMuteEvents.FirstOrDefault(x => x.Unmuted == false);
+        }
+
+        private IEnumerable<MuteEvent> GetMuteEvents(DiscordServerContext server, ulong userId)
+        {
+            var getMuteEvents = new GetMuteEventsFromBaseQuery(server.Id);
+            var allServerMuteEvents = _queryBus.Execute(getMuteEvents).MuteEvents;
+
+            var userMuteEvents = allServerMuteEvents.Where(x => x.UserId == userId);
+            return userMuteEvents;
+        }
+
+        private async void UnmuteInFuture(Contexts contexts, MuteEvent muteEvent, UserContext userToUnmute)
+        {
+            var timeRange = muteEvent.TimeRange;
+            await Task.Delay(timeRange.End - timeRange.Start);
+
+            if (IsStillMuted(contexts.Server, muteEvent))
+            {
+                UnmuteUserOnlyIfMuted(contexts, userToUnmute);
+            }
+        }
+
+        private bool IsStillMuted(DiscordServerContext server, MuteEvent muteEvent)
```

Comment:
postaraj sie to przenieść do serwisu, ale tak żeby metody tego serwisu dostawały pełne dane
tzn żeby nie musiały sie odwoływać do bazy 

np tu w środku robisz GetMuteEvents
a lepiej gdybyś przekazywał te eventy jako parametr metody